ComplementaryScripts/modelCuration/mapKEGGID.m

```diff
@@ -0,0 +1,275 @@
+                end
+            end
+        end
+    end
+end
+
+if size(xref_rxnMetaCyc,2) == 5
+    %only retain mapped rxnKEGGID if model.rxnKEGGID is empty
+    empties = find(cellfun('isempty',xref_rxnMetaCyc(:,5)));
+    xref_rxnMetaCyc(empties,:) = [];
+    empties = find(~cellfun('isempty',xref_rxnMetaCyc(:,4)));
+    xref_rxnMetaCyc(empties,:) = [];
+    
+    %add rxnKEGGID into model
+    [~,idx] = ismember(xref_rxnMetaCyc(:,2),model.rxns);
+    idx = idx(idx~=0);
+    
+    for i = 1:length(idx)
+        model.rxnKEGGID(idx(i)) = xref_rxnMetaCyc(i,5);
+    end
+else
+    xref_rxnMetaCyc(:) = []; %erase all cell array, since no KEGGID matched and added
+end
+
+%map rxnMetaNetXID to rxnSEEDID via function mapIDsViaMNXref, then map rxnSEEDID to rxnKEGGID
+xref_rxnSEED = mapIDsViaMNXref('rxns',model.rxnMetaNetXID,'MetaNetX','SEED');
+xref_rxnSEED(:,2) = model.rxns; %match rxnSEEDID with model.rxns
+xref_rxnSEED(:,3) = model.rxnMetaNetXID; %match rxnSEEDID with model.rxnMetaNetXID
+xref_rxnSEED(:,4) = model.rxnKEGGID; %match rxnSEEDID with model.rxnKEGGID
+empties = find(cellfun('isempty',xref_rxnSEED(:,1)));
+xref_rxnSEED(empties,:) = [];
+
+%load reactions.tsv from modelSEEDDatabase
+temp = what('ModelSEEDDatabase\Biochemistry'); %require cloned/downloaded GitHub repo ModelSEEDDatabase to be added to MATLAB path
+cd(temp.path);
+fid2 = fopen('reactions.tsv');
+format = repmat('%s ',1,22);
+format = strtrim(format);
+rxn_temp = textscan(fid2,format,'Delimiter','\t','HeaderLines',0);
+for i = 1:length(rxn_temp)
+    SEED_rxnInfo(:,i) = rxn_temp{i};
+end
+commentLines = startsWith(SEED_rxnInfo(:,1),'#');
+SEED_rxnInfo(commentLines,:) = [];
+fclose(fid2);
+
+%Input rxnKEGGID into xref array
+for i = 1:length(xref_rxnSEED)
+    if isempty(xref_rxnSEED{i,4})
+        idx = find(ismember(SEED_rxnInfo(:,1),xref_rxnSEED(i,1)));
+        if idx~=0
+            if ~isempty(SEED_rxnInfo{idx,13}) && contains(SEED_rxnInfo(idx,13),'KEGG')
+                temp_ID = extractBetween(SEED_rxnInfo(idx,13),'KEGG: ','|');
+                xref_rxnSEED(i,5) = temp_ID;
+            end
+        end
+    end
+end
+
+if size(xref_rxnSEED,2) == 5
+    %only retain mapped rxnKEGGID if model.rxnKEGGID is empty
+    empties = find(cellfun('isempty',xref_rxnSEED(:,5)));
+    xref_rxnSEED(empties,:) = [];
+    empties = find(~cellfun('isempty',xref_rxnSEED(:,4)));
+    xref_rxnSEED(empties,:) = [];
+    
+    %add rxnKEGGID into model
+    [~,idx] = ismember(xref_rxnSEED(:,2),model.rxns);
+    idx = idx(idx~=0);
+    
+    for i = 1:length(idx)
+        model.rxnKEGGID(idx(i)) = xref_rxnSEED(i,5);
+    end
+else
+    xref_rxnSEED(:) = []; %erase all cell array, since no KEGGID matched and added
+end
+
+%% metKEGGID
+%map metMetaNetXID to metKEGGID via function mapIDsViaMNXref
+temp = what('RAVEN\external\MetaNetX'); %require switching to branch: feat/add_MetaNetX and adding RAVEN to MATLAB directory (to modify after feature branch is merged)
```

Comment:
If you are using the `mapIDsViaMNXref` in yeastGEM, then it is not required to switch branch manually. Switching branch will be done automatically in the script `mapIDsViaMNXref`

Reply:
Resolved in b8a5d3283c45e15a654eb5e89e91570a7406ff61.